kyuubi-server/src/main/scala/org/apache/kyuubi/engine/ProcBuilder.scala

```diff
@@ -148,6 +149,38 @@ trait ProcBuilder {
     proc
   }
 
+  def execKillYarnJob(appId: String): Int = {
+    val hadoopHomeOpt = env.get("HADOOP_HOME").orElse {
+      throw KyuubiSQLException(s"Failed to kill yarn job. HADOOP_HOME is not set! " +
+        "For more detail information on installing and configuring Yarn, please visit " +
+        "https://kyuubi.apache.org/docs/stable/deployment/on_yarn.html")
+    }.map { hadoopHome =>
+      Paths.get(hadoopHome, "bin", "yarn").toAbsolutePath.toFile.getCanonicalPath
+    }.get
+    val command = hadoopHomeOpt + " application -kill " + appId
```

Comment:
we can use yarn rest api to kill

Reply:
This way makes more sense to me, because it's doesn't require setting up hadoop client in Kyuubi machine, and specific versions of hadoop client jars.